spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/SpannerKeyIdConverter.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.spanner;
+
+import java.io.Serializable;
+import java.util.StringJoiner;
+
+import com.google.cloud.spanner.Key;
+
+import org.springframework.data.rest.webmvc.spi.BackendIdConverter;
+
+/**
+ * @author Balint Pato
+ */
+public class SpannerKeyIdConverter implements BackendIdConverter {
+
+	@Override
+	public Serializable fromRequestId(String id, Class<?> entityType) {
+		Object[] parts = id.split(getUrlIdSeparator());
+		return Key.of(parts);
+	}
+
+	protected String getUrlIdSeparator() {
+		return "_";
```

Comment:
But isn't this also a totally legit character to have in a STRING column in Spanner that can a key column? Wouldn't that mean this has the same problem as the comma?